src/lib/game/world/content-queue.js

```diff
@@ -0,0 +1,83 @@
+class ContentQueue {
+
+  constructor(processor, interval = 1, workFactor = 1, minWork = 1) {
```

Comment:
The idea is to yield back to anything else that's been waiting while the queue items were being processed (ie cooperative multitasking). I suppose the value could even be ```0```, and not configured. ```0``` seemed like it might be relying a bit heavily on convention in JS VMs to always yield on ```setTimeout```, though.